src/network/networkd-json.c

```diff
@@ -1036,6 +1037,52 @@ static int dhcp_server_append_json(Link *link, JsonVariant **v) {
         return json_variant_set_field_non_null(v, "DHCPServer", w);
 }
 
+static int dhcp6_get_vendor_options_append_json(Link *link, sd_dhcp6_option *option, JsonVariant **ret) {
+
+        assert(link);
+        assert(ret);
+
+        if (!link->dhcp6_lease)
+                return 0;
+
+        return json_variant_append_arrayb(ret,
+                                       JSON_BUILD_OBJECT(
+                                                       JSON_BUILD_PAIR_UNSIGNED("EnterpriseId", option->enterprise_identifier),
+                                                       JSON_BUILD_PAIR_UNSIGNED("SubOptionCode", option->option),
+                                                       JSON_BUILD_PAIR_HEX("SubOptionData", option->data, option->length)));
+}
+
+static int dhcp6_client_vendor_options_append_json(Link *link, JsonVariant **v) {
+        _cleanup_(json_variant_unrefp) JsonVariant *array = NULL;
+        sd_dhcp6_option **options = NULL;
+        size_t n_vendor_options;
+        int r;
+
+        assert(link);
+        assert(v);
+
+        if (!link->dhcp6_lease)
+                return 0;
+
+        r = set_dump_sorted(link->dhcp6_lease->vendor_options, (void***) &options, &n_vendor_options);
+        if (r < 0)
+                return r;
+
+        FOREACH_ARRAY(option, options, n_vendor_options) {
+                _cleanup_(json_variant_unrefp) JsonVariant *e = NULL;
+
+                r = dhcp6_get_vendor_options_append_json(link, *option, &e);
+                if (r < 0)
+                        return 0;
+
+                r = json_variant_append_array(&array, e);
```

Comment:
This is redundant. Please pass `array` to `json_variant_append_arrayb()`.

Reply:
Removed